greet_app/greet.py

```diff
@@ -1,4 +1,10 @@
 def greet(name):
     if name is None:
         name = "my friend"
+    if isinstance(name, str):
+        if name.isupper():
+            return "HELLO {}!".format(name)
```

Comment:
since you are on python 3 let's use f-string as they are less verbose https://docs.python.org/3/reference/lexical_analysis.html#f-strings